packages/discovery-react-components/src/components/DocumentPreview/utils/documentData.ts

```diff
@@ -42,7 +43,10 @@ export function isJsonFile(doc: QueryResult | null | undefined): boolean {
 /**
  * Returns the preview type for document
  */
-export function detectPreviewType(document: DiscoveryDocument, file?: string): PreviewType {
+export function detectPreviewType(
+  document: DiscoveryDocument,
+  file?: string | PDFSource
```

Comment:
We are often repeating `string | PDFSource` type. We should alias that to a type named `DocumentFile` or something like that. Then, later, if we add another supported type, we only need to update the definition.

Reply:
Thank you for pointing this out. It made the code much clean. As we accept `ArrayBuffer` as well, the type is `string | ArrayBuffer | PDFSource`